jax/test_util.py

```diff
@@ -868,3 +871,79 @@ def ignore_warning(**kw):
   with warnings.catch_warnings():
     warnings.filterwarnings("ignore", **kw)
     yield
+
+
+class _cached_property:
+  null = object()
+
+  def __init__(self, method):
+    self._method = method
+    self._value = self.null
+
+  def __get__(self, obj, cls):
+    if self._value is self.null:
+      self._value = self._method(obj)
+    return self._value
+
+
+class _LazyDtypes:
+  """A class that unifies lists of supported dtypes.
+
+  These could be module-level constants, but the device is not always
```

Comment:
```suggestion
  These could be module-level constants, but device_under_test() is not always
```

Maybe more specific is better?